### PR TITLE
add translation system to pygeoapi

### PR DESCRIPTION
**Pull request by pvgenuchten**
this pr adds a translation system to pygeoapi, see locale/readme.md on how to set up

- extract latest keys in .pot file with: `pybabel extract -F babel-mapping.ini -o locale/messages.pot ./`
- then update .po language file: `pybabel update -d locale -l fr -i locale/messages.pot`
- update translations and compile .mo file: `pybabel compile -d locale -l fr`

Run pygeoapi and switch language to french ?lang=fr, notice some hardcoded strings translated.

### CODE REVIEW

File: pygeoapi/util.py
Comment (GeoSander) [inline pygeoapi/util.py:350]: We might need to check how efficient this is (loading all translations each time when `render_j2_template` is called). If the locale has not changed, there's no need to reload. But perhaps `install_gettext_translations` already has such a mechanism in place.

File: pygeoapi/util.py
Comment (GeoSander) [inline pygeoapi/util.py:46]: Please remove this line :)

**Review by GeoSander — APPROVED** [review]
Good start @pvgenuchten and happy to see that it was easy to put the translations in place!
For the links sections (as generated by `api.py`) we should have a look how to properly build in the Babel/gettext stuff there, but it might be better to do this after the API refactor, don't you agree @tomkralidis?

File: locale/fr/LC_MESSAGES/messages.po
Comment (alexandreleroux) [inline locale/fr/LC_MESSAGES/messages.po:56]: License in French is spelled `Licence`, thanks

File: pygeoapi/util.py
Comment (tomkralidis) [reply pygeoapi/util.py:350]: > Good start @pvgenuchten and happy to see that it was easy to put the translations in place!
> For the links sections (as generated by `api.py`) we should have a look how to properly build in the Babel/gettext stuff there, but it might be better to do this after the API refactor, don't you agree @tomkralidis?

+1

File: locale/readme.md
Comment (tomkralidis) [inline locale/readme.md:1]: Rename file to `locale/README.md`

**Review by tomkralidis — CHANGES_REQUESTED** [review]
Great work here @pvgenuchten !  Minor comments.  I would also add a sentence or two in https://docs.pygeoapi.io/en/latest/language.html

**Review by tomkralidis — APPROVED** [review]

**Comment by GeoSander** [comment]
@tomkralidis is it okay to merge this (after rebase)?

**Comment by tomkralidis** [comment]
@GeoSander +1, let me know when rebased.

**Comment by pvgenuchten** [comment]
just did a rebase

**Pull request merged**
